db/revision.go

```diff
@@ -22,13 +23,35 @@ import (
 type Body map[string]interface{}
 
 const (
-	BodyDeleted = "_deleted"
-	BodyRev     = "_rev"
-	BodyId      = "_id"
+	BodyDeleted   = "_deleted"
+	BodyRev       = "_rev"
+	BodyId        = "_id"
+	BodyRevisions = "_revisions"
+)
+
+// A revisions property found within a Body.  Expected to be of the form:
+//   Revisions["start"]: int64, starting generation number
+//   Revisions["ids"]: []string, list of digests
+// Used as map[string]interface{} instead of Revisions struct because it's unmarshalled
+// along with Body, and we don't need the overhead of allocating a new object
+type Revisions map[string]interface{}
+
+const (
+	RevisionsStart = "start"
+	RevisionsIds   = "ids"
 )
 
 func (b *Body) Unmarshal(data []byte) error {
-	if err := json.Unmarshal(data, &b); err != nil {
+
+	if len(data) == 0 {
+		b = &Body{}
```

Comment:
This isn't going to do what I think the intention was.
It will change the value of b, but only in this scope (and not for the caller of `b.Unmarshal`)

```go
*b = Body{}
```

The test in the next comment reproduces this

Reply:
Should actually be returning error here, to match json.Unmarshal behaviour.  Fixed and added to unit test.